src/oscar/apps/dashboard/ranges/forms.py

```diff
@@ -40,43 +42,47 @@ def clean(self):
         if not clean_data.get('query') and not clean_data.get('file_upload'):
             raise forms.ValidationError(
                 _("You must submit either a list of SKU/UPCs or a file"))
-        return clean_data
-
-    def clean_query(self):
-        raw = self.cleaned_data['query']
-        if not raw:
-            return raw
-
-        # Check that the search matches some products
-        ids = set(UPC_SET_REGEX.findall(raw))
-        products = self.range.all_products()
-        existing_skus = set(products.values_list(
-            'stockrecords__partner_sku', flat=True))
-        existing_upcs = set(products.values_list('upc', flat=True))
-        existing_ids = existing_skus.union(existing_upcs)
-        new_ids = ids - existing_ids
-
-        if len(new_ids) == 0:
-            raise forms.ValidationError(
-                _("The products with SKUs or UPCs matching %s are already in"
-                  " this range") % (', '.join(ids)))
+        raw = clean_data['query']
```

Comment:
It was not possible to add the changes in clean_query, because I could not access `clean_data['upload_type']` there. Otherwise I would have to call super.clean() twice. Any ideas?

Reply:
ah, misunderstood... fixed.